Contributing (#544)
api3dao/vitepress-docs: docs/explore/introduction/contributing.md

<FlexStartTag/>

# {{$frontmatter.title}} 

We think critical infrastructure, like API3 DAO's first-party oracles, should

> implement open and transparent contributor processes to align to our vision of

Reviewer: ```suggestion
implement open and transparent contributor processes to align with our vision of
```

being a programmatically driven protocol.

To support DAO governance and inform voters, all teams publish a public proposal
at regular intervals, meaning strategy & operations are accessible on the
[API3 Governance Forum](https://forum.api3.org/) for all to understand.

> Additionally, teams generally update the community with retrospective reports.

Reviewer: ```suggestion
Additionally, teams generally update the community with ongoing reports.
```

## Types of contributors

Community participants within a Decentralized Autonomous Organization (DAO) have
an active role in shaping the future of the project. Our aim is to simplify the
process for those who wish to participate, providing predefined methods for

> engagement. Generally these take forum in three routes:

Reviewer: ```suggestion
engagement. Generally these take three forms:
```

If contributors feel they possess particular skills or expertise that may
benefit the proposal's execution, they can directly communicate with the
proposal leads. This collaborative approach encourages a diverse array of
viewpoints and fosters a proactive culture where everyone's skills and
experiences are leveraged for the benefit of the DAO's objectives.

### Take on pre-defined bounties

Community participants have the opportunity to engage with pre-set bounties,
organized in a modular structure to offer clear and attainable tasks. For
instance, a bounty might be created for a specific job that is expected to be
completed within two days. This structure allows any member of the community to
take on the job, complete it, and subsequently receive the designated grant as a
reward retrospectively.

### Submit your own proposal

We welcome proposals that bring new ideas or sub-teams to the DAO. Generally
engaging the DAO through our community in a constructive manner will help
sentiment check a proposal and help shape it to align to current contributors.
If you are proposing an initiative that would require collaboration with a

> current sub team, we suggest co-ordinating proactively and outlining how you

Reviewer: ```suggestion
current sub team, we suggest coordinating proactively and outlining how you
```

will work together.

# Put forward your ideas to the API3 DAO

Regardless of the contributor's skill, API3's vision is for participation to be
as frictionless as possible. Individuals with a relevant background can
contribute to the project in a flexible manner, replicating similar

> freelancer/consultancy type figures commonly seen.

Reviewer: ```suggestion
freelancer/consultancy type models commonly seen.
```

## Technical contributors

For technical folks this could be through development, integrations, writing
documentation, etc. Although the general challenge is that one needs to be able
to use a solution proficiently before being able to improve it. If this applies

> we suggest digesting the get started materials.

Reviewer: A link would be helpful for "getting started"

Author: updated

To begin with, this contribution may look like:

- opening issues
- making pull requests
- proposing features

### To get started

We recommend spending some time learning about API3 DAO's oracle technology
before looking to contribute from an engineer perspective. We welcome you to ask
any questions on [Discord<ExternalLinkImage/>](https://discord.gg/qnRrcfnm5W).

- Read the entirety of these docs. Take notes and cross-reference, but do not
  expect to understand everything. Feel free to ask questions.
- Visit the [API3 GitHub<externalLinkImage/>](https://github.com/api3dao) page

>   and find everything about.. everything. Explore the many repositories to gain

Reviewer: I suggest rewording this line.

  in-depth knowledge about API3.
- Go through the issues, PRs, etc. and understand the current focus of the
  development team by viewing the development board.

### If you are learning about blockchain & oracles

- Read the
  [getting started<ExternalLinkImage/>](docs/explore/introduction/index.md)
  section in the API3 Docs
- (For the curious) Explore the API3 codebase on
  [GitHub<ExternalLinkImage/>](https://github.com/api3dao)
- If you do not have any blockchain development experience, read
  [Mastering Ethereum<ExternalLinkImage/>](https://github.com/ethereumbook/ethereumbook)
  cover to cover. It is a light read that touches on all subjects that you will
  need to know about superficially.

## Non-technical contributors

Marketers, creatives and community builders can contribute to DAO subteams with

> ad-hoc projects or to support areas that are lacking focus. The challenge here

Reviewer: I'd suggest rewording 'are lacking focus'

Author: Re worked area a little, thanks

delivering in alignment with other involved contributors.

Contirbutions may look like:

- blog posts, community resources
- ambassador activity, setting of events
- art or branded merch

# How do I access funds to support my efforts? 

Prospective contributors should provide a clear scope of work with (1) detailed
deliverables and estimations of completion time (2) any applicable coordination
with other contributors necessary to accomplish the deliverables and (3) the
amount of API3 tokens requested.

We appreciate the individual nature of contributing and these frameworks are
flexible, yet linking these activities to the challenges outlined in team
proposals will support our hollisitic objectives and be more likely supported.

## Ecosystem fast bounties & retropective grants

The API3 Ecosystem team has a dedicated pool of funds to support micro-grants from community contributors. This can be retrospective or pre-planned, but has been created to make rewarding contribution frictionless and simple. 

## This can be submitted in a variety of way

- Join the API3 [community<ExternalLinkImage/>](https://discord.gg/qnRrcfnm5W)

> and submit within the ecosystem contributor request channel. 

Reviewer: This channel doesn't exist?

- Submit an

> overview onto the [API3 forums](https://forum.api3.org/). 

Reviewer: I would suggest using "sentiment check" instead of overview, and/or linking to https://docs.api3.org/guides/dao-members/proposals.html#proposal-creation

to use a solution proficiently before being able to improve it. If this applies
we suggest digesting the [get started materials](index.md).

To begin with, this contribution may look like:

- opening issues
- making pull requests
- proposing features

### To get started

We recommend spending some time learning about API3 DAO's oracle technology
before looking to contribute from an engineer perspective. We welcome you to ask
any questions on [Discord<ExternalLinkImage/>](https://discord.gg/qnRrcfnm5W).

- Read the entirety of these docs. Take notes and cross-reference, but do not
  expect to understand everything. Feel free to ask questions.
- Visit the [API3 GitHub<externalLinkImage/>](https://github.com/api3dao) page
  to explore the many repositories of API3's oracle stack and gain in-depth
  knowledge.
- Head to the
  [API3 Community Hub](https://github.com/api3-ecosystem/doc-hub.git) with get
  started tutorials & demo projects.
- Go through the issues, PRs, etc. and understand the current focus of the
  development team by viewing the development board.

### If you are learning about blockchain & oracles

- Read the

>   [getting started<ExternalLinkImage/>](docs/explore/introduction/index.md)

Reviewer: 404 currently.
```suggestion
  [getting started<ExternalLinkImage/>](index.md)
```

Reviewer 2: Use: `/explore/introduction/index.md`

  [Mastering Ethereum<ExternalLinkImage/>](https://github.com/ethereumbook/ethereumbook)
  cover to cover. It is a light read that touches on all subjects that you will
  need to know about superficially.

## Non-technical contributors

Marketers, creatives and community builders can contribute to DAO with ad-hoc
projects or to support specific areas within subteams. The challenge here is
accurately scoping, forming a brief proposed solution method, and then
delivering in alignment with other involved contributors.

Amoung many other types, contirbutions may look like:

- blog posts, community resources
- ambassador activity, setting of events
- graphic design, creative
- testers, user research

## How do I access funds to support my efforts?

Prospective contributors should provide a clear scope of work with (1) detailed
deliverables and estimations of completion time (2) any applicable coordination
with other contributors necessary to accomplish the deliverables and (3) the
amount of API3 tokens requested.

We appreciate the individual nature of contributing and these frameworks are
flexible, yet linking these activities to the challenges outlined in team

> proposals will support our hollisitic objectives and be more likely supported.

Reviewer: spellcheck

> ### Ecosystem fast bounties & retropective grants

Reviewer: spellcheck

The API3 Ecosystem team has a dedicated pool of funds to support micro-grants
from community contributors. This can be retrospective or pre-planned, but has
been created to make rewarding contribution frictionless and simple.

### This can be submitted in a variety of way

- Join the API3 [community<ExternalLinkImage/>](https://discord.gg/qnRrcfnm5W)
  and submit within the ecosystem contributor request channel.
- Submit an overview and sentiment check onto the
  [API3 forums](https://forum.api3.org/).
- Follow the
  [proposal guidelines](https://docs.api3.org/guides/dao-members/proposals.html#proposal-creation)
  provided by API3 DAO.
- Email ecosystem@api3.org with an overview

Within this process we welcome open dialogue and as a DAO recongize that

> external contributors and diversity in thought is a strenght.

Reviewer: spellcheck & same on line above

> ## Non-technical contributors

Reviewer: I think you want 3 # here

Marketers, creatives and community builders can contribute to DAO with ad-hoc
projects or to support specific areas within subteams. The challenge here is
accurately scoping, forming a brief proposed solution method, and then
delivering in alignment with other involved contributors.

> Amoung many other types, contirbutions may look like:

Reviewer: Please use spellcheck

flexible, yet linking these activities to the challenges outlined in team
proposals will support our hollisitic objectives and be more likely supported.

### Ecosystem fast bounties & retropective grants

The API3 Ecosystem team has a dedicated pool of funds to support micro-grants
from community contributors. This can be retrospective or pre-planned, but has
been created to make rewarding contribution frictionless and simple.

> ### This can be submitted in a variety of way

Reviewer: I suggest rewording this section title
